### ISSUE TITLE
Blazor "dotnet watch run" and "VS 2019 Preview" issues with .NET 5.0.1 update

### ISSUE THREAD

**Issue by majorimi** [dotnet/aspnetcore#28541]
I have a Blazor project which was working fine. Few hours ago updated VS 2019 to the latest "Version 16.9.0 Preview 2.0". I also updated `"Microsoft.AspNetCore.Components.WebAssembly" Version="5.0.1"` Nuget package. I have done several Clean, Rebuild even restarted VS 2019. But still getting the same outcome. 

1. Newly opened VS instance with Blazor project builds correctly no messages in the "Error List" window. `dotnet watch run` command works. Site opened in the browser.
2. Smallest code change (only added a space) will result nearly 100 build errors in the "Error List" window, meanwhile got "Build succeeded" status message. Also if I start the application from VS (press F5) it runs. **But app works correctly in case of Server hosted mode. Blazor Web Assembly app looks very weird. It seems isolated CSS was not collected for the app.**
3. Already running 'watch' detects code change and also rebuilds the code with the error below:

Error in `dotnet watch run` console:
`C:\Program Files\dotnet\sdk\5.0.200-preview.20601.7\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error : The given key 'tfm_raw' was not present in the dictionary.`

**As I know .NET 5.0.1 update was released recently. But I think it should be a "final" package not a preview?** Because checking installed apps on my computer VS was installed a Preview package.

![image](https://user-images.githubusercontent.com/7632824/101648305-f7362600-3a39-11eb-86c7-1adebee301af.png)

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us. This looks like caused by the preview SDK you're using. That is being installed by the preview version of VS. 
@marcpopMSFT do you know what may cause this error?

**Comment by marcpopMSFT** [comment]
The VS preview is indeed what installed the 5.0.200 preview of the sdk. That VS preview would have come with the 5.0.0 runtime and your separate install of 5.0.1 should be fine.  That error is coming from NuGet so including @zkat and @aortiz-msft in case they know of a breaking change from NuGet that got picked up in that preview maybe.

**Comment by majorimi** [comment]
Yes after reporting this issue I have installed manually the new .Net 5.0.1 SDK. Not much improved though.. Still getting this error. 
But I think it is still very weird VS installs a preview version of the SDK. Never seen this before.

**Comment by zkat** [comment]
There's been a PR for this for a while, but some stuff came up and it didn't get merged into our preview branch: https://github.com/NuGet/NuGet.Client/pull/3730

**Comment by majorimi** [comment]
Ok but **before this update `dotnet watch run` command was wondering perfectly fine.** After yesterday's update I'm getting the explained weird behavior: **watch console build error on e.g. a space change and app cannot be r freshed, also VS built errors meanwhile app can be run with F5**.

I would say it's a bug. Since different tools show different build results for exactly the same codebase. Supposedly broken code can be run. But in Blazor Web Assembly mode UI is broken.... So Blazor development now is nearly impossible.

No one else has this problem? Is there any suggestion how to remove/disable this Nuget package or the installed preview .Net 5 SDK?

**Comment by zkat** [comment]
No it's definitely a bug, don't get me wrong! I'm gonna work on getting this PR merged ASAP so we can get you unblocked :)

**Comment by majorimi** [comment]
Thanks @zkat 
Hope it will be quick. Becasuese I'm working on lots of [Blazor components](https://github.com/majorimi/blazor-components). But now Blazor development is quite painful...

**Comment by aortiz-msft** [comment]
@majorimi - Is it possible for you to revert back to Visual Studio 16.8 to get unblocked in the meantime?

**Comment by majorimi** [comment]
@aortiz-msft  As I read about it "revert" is full uninstall and do a new install. Which is quite a big process... But even I'm not able to find older installers since [this page ](https://docs.microsoft.com/en-us/visualstudio/releases/2019/history#release-dates-and-build-numbers) was not updated for some time.

**Comment by aortiz-msft** [comment]
@majorimi - Which version were you hoping to revert to? The page has a link to the latest publicly released version which is 16.8.3

**Comment by MarkStega** [comment]
I presume that this error is the same issue (I am also on VS Version 16.9.0 Preview 2.0):
```
C:\Solutions\OHI\Optimiser>dotnet pack --configuration Release_Server --output "c:\solutions\Local Nuget Packages\GC" -p:InformationalVersion=2.0.0-aaa1 GeneralComponents\GeneralComponents.csproj
Microsoft (R) Build Engine version 16.8.0+126527ff1 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\Solutions\OHI\Optimiser\GeneralComponents\GeneralComponents.csproj (in 427 ms).
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  GeneralComponents -> C:\Solutions\OHI\Optimiser\GeneralComponents\bin\Release_Server\net5.0\GeneralComponents.dll
C:\Program Files\dotnet\sdk\5.0.200-preview.20601.7\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error : The given key 'tfm_raw' was not present in the dictionary. [C:\Solutions\OHI\Optimiser\GeneralComponents\GeneralComponents.csproj]
```

**Comment by zkat** [comment]
Yup! Same one

**Comment by MarkStega** [comment]
@majorimi Here is a much simpler reversion:
Get the 5.0.101 SDK from [https://dotnet.microsoft.com/download/dotnet/5.0](https://dotnet.microsoft.com/download/dotnet/5.0)
and add a global.json with:
```
{
    "sdk": {
        "version": "5.0.101",
        "rollForward": "disable"
    }
}
```

**Comment by majorimi** [comment]
I have installed a stable version of Visual Studio 2019 which works with my project. It does not solve the issue with **Preview** instance. At least now I'm not blocked.

**Comment by zkat** [comment]
@majorimi the patch has been merged into the appropriate preview branches for VS and .NET SDK, so this should be fixed with the next preview. :)

**Comment by mkArtakMSFT** [comment]
Thanks @zkat. Closing this issue as resolved then. In the future, we should move issues to the appropriate repo (NuGet in this case  as I understood).

**Comment by toreaurstadboss** [comment]
> @majorimi Here is a much simpler reversion:
> Get the 5.0.101 SDK from https://dotnet.microsoft.com/download/dotnet/5.0
> and add a global.json with:
> 
> ```
> {
>     "sdk": {
>         "version": "5.0.101",
>         "rollForward": "disable"
>     }
> }
> ```
Great now it works :)
